static/js/components/tiles/histogram_tile.tsx

```diff
@@ -64,70 +72,123 @@ function getLastDayOfMonth(dateString: string): string {
  */
-function getMonthsArray(dateSetting: string): string[] {
+function getLabelBounds(dateSetting: string): [string, string] {
   // get start and end of dates to show data for
   let [startDate, endDate] = getDateRange(dateSetting);
 
   // specify full dates if start and end dates are just YYYY or YYYY-MM
-  if (startDate.length == "YYYY".length && endDate.length == "YYYY".length) {
+  if (
+    startDate.length == YEAR_FORMAT.length &&
+    endDate.length == YEAR_FORMAT.length
+  ) {
     startDate = `${startDate}-01-01`;
     endDate = `${endDate}-12-31`;
   } else if (
-    startDate.length == "YYYY-MM".length &&
-    endDate.length == "YYYY-MM".length
+    startDate.length == MONTH_FORMAT.length &&
+    endDate.length == MONTH_FORMAT.length
   ) {
     startDate = `${startDate}-01`;
     endDate = `${getLastDayOfMonth(endDate)}`;
   }
+  return [startDate, endDate];
+}
+
+/**
+ * Helper function for getting all days along x-axis to display.
+ * Used for initializing bins when binning daily.
+ * @param dateSetting user selected date setting
+ */
+function getDaysArray(dateSetting: string): string[] {
+  const [startDate, endDate] = getLabelBounds(dateSetting);
+
+  // Fill in days between start and end dates
+  const days = new Array<string>();
+  for (
+    let date = new Date(startDate + "Z");
+    getFormattedDate(date, DAY_FORMAT) <=
+    getFormattedDate(new Date(endDate + "Z"), DAY_FORMAT);
+    date.setUTCDate(date.getUTCDate() + 1)
+  ) {
+    days.push(getFormattedDate(new Date(date), DAY_FORMAT));
+  }
+  return days;
+}
+
+/**
+ * Helper function for getting all months along x-axis to display.
+ * Used for initializing bins when binning monthly.
+ * @param dateSetting user selected date setting
+ */
+function getMonthsArray(dateSetting: string): string[] {
+  const [startDate, endDate] = getLabelBounds(dateSetting);
 
   // Fill in months between start and end dates
   const months = new Array<string>();
   for (
     let date = new Date(startDate + "Z");
-    getMonthString(date) <= getMonthString(new Date(endDate + "Z"));
+    getFormattedDate(date, MONTH_FORMAT) <=
+    getFormattedDate(new Date(endDate + "Z"), MONTH_FORMAT);
     date.setMonth(date.getMonth() + 1)
   ) {
-    months.push(getMonthString(new Date(date)));
+    months.push(getFormattedDate(new Date(date), MONTH_FORMAT));
   }
   return months;
 }
 
 /**
- * Helper function to bin data by month
+ * Helper function for binning data.
+ * @param disasterEventPoints event data points to bin
+ * @param dateSetting user selected date setting
+ * @param format temporal granularity to use. One of "YYYY-MM" or "YYYY-MM-DD"
  */
-function binDataByMonth(
+function binData(
   disasterEventPoints: DisasterEventPoint[],
-  dateSetting: string
+  dateSetting: string,
+  format: string
 ): DataPoint[] {
   if (_.isEmpty(disasterEventPoints)) {
     return [];
   }
 
-  // Track YYYY-MM -> number of events
+  // Track YYYY-MM / YYYY-MM-DD -> number of events
```

Comment:
nit: just "Track date -> number of events"

Reply:
changed to "date"